### PR TITLE
Add public ip address as a parameter in New-AzRouteServer

### PR DESCRIPTION
**Pull request by litchiyangMSFT**
## Description

Customers need to input public ip address as a mandatory parameter to create route server

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/RouteServer/NewAzureRMRouteServerCommand.cs [hunk 1]
```diff
@@ -50,6 +50,12 @@ public partial class NewAzureRmRouteServer : RouteServerBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string HostedSubnet { get; set; }
 
+        [Parameter(
+            Mandatory = true,
```
Comment (BethanyZhou) [inline src/Network/Network/RouteServer/NewAzureRMRouteServerCommand.cs:54]: adding a new mandatory parameter will cause a breaking change.
Where is your design review?

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Network/Network/RouteServer/NewAzureRMRouteServerCommand.cs (hunk 1 above)
Comment (litchiyangMSFT) [reply src/Network/Network/RouteServer/NewAzureRMRouteServerCommand.cs:54]: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/969

Comment (litchiyangMSFT) [reply src/Network/Network/RouteServer/NewAzureRMRouteServerCommand.cs:54]: Please review. Thanks

**Comment by BethanyZhou** [comment]
Since you are developing network-related feature, are you confirmed which branch this PR should go?

**Comment by VeryEarly** [comment]
As discussed offline, this feature do not depend on new API version (should be fine to target to main branch)
However extra work needed at end of current milestone:
https://github.com/Azure/azure-powershell-pr/wiki/Sync-Protected-Branch

Please also remove "Mandatory" for new parameter.

File: src/Network/Network/help/New-AzRouteServer.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ Creates an Azure RouteServer.
 
 ```
 New-AzRouteServer -ResourceGroupName <String> -RouteServerName <String> -HostedSubnet <String>
- -Location <String> [-Tag <Hashtable>] [-Force] [-AsJob] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
- [-Confirm] [<CommonParameters>]
+ -PublicIpAddress <PSPublicIpAddress> -Location <String> [-Tag <Hashtable>] [-Force] [-AsJob]
```
Comment (BethanyZhou) [inline src/Network/Network/help/New-AzRouteServer.md:17]: Please update help file as PublicIpAddress is optional now.

File: src/Network/Network/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,8 @@
 --->
 
 ## Upcoming Release
+* Added public ip address as a mandatory parameter to create route server
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:22]: Please update change log as PublicIpAddress is optional now.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

**Comment by BethanyZhou** [comment]
Please check the scenario test TestRouteServerPeerCRUD since it's failed

File: src/Network/Network/help/New-AzRouteServer.md (hunk 2 above)
Comment (litchiyangMSFT) [reply src/Network/Network/help/New-AzRouteServer.md]: updated

File: src/Network/Network/ChangeLog.md (hunk 3 above)
Comment (litchiyangMSFT) [reply src/Network/Network/ChangeLog.md]: updated

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**